### PR TITLE
chore: cleanup github actions workflows

### PR DESCRIPTION
**Pull request by shaneutt**
This patch combines all testing into a single workflow and
action, as well as coverage since this is all instrumented
with a single Makefile directive already anyway. Additionally
this patch flags the release workflow to wait for tests to
be passing on main before committing to a release of a tag.

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -27,10 +27,9 @@ jobs:
           ${{ runner.os }}-go
     - name: checkout repository
       uses: actions/checkout@v2
-    - name: run integration tests
+    - name: run unit and integration tests
       run: make test.integration
```
Comment (mflendrich) [inline .github/workflows/tests.yaml:31]:
```suggestion
      run: make test.unit test.integration
```

Given that you're removing a workflow that `make`s `test.unit`, don't we need to add `test.unit` here now that this step is supposed to run those?

**Review by mflendrich — APPROVED** [review]
LGTM blocked by resolution of the attached comment

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (shaneutt) [reply .github/workflows/tests.yaml:31]: We don't, the integration tests in this makefile run ALL tests, simply _including_ the integration tests (by way of go tag).

Comment (mflendrich) [reply .github/workflows/tests.yaml:31]: That's a misleading name for that target, then. Consider renaming to `test.all` alongside `test.integration` and `test.unit`

Comment (shaneutt) [reply .github/workflows/tests.yaml]: Sounds good, 09c27b175fd81ae78f7ebefa9bdbc36e5881b884 adds this :+1:

**Pull request merged**